### PR TITLE
Generate init

### PR DESCRIPTION
**Pull request by Yanqiao4396**
# Set initial modules of generate file

## Description

Create the initial modules the subsequent tasks should all follow. Left the necessary comments and docstrings to help create behavior.

## Linked Issues

closes: #44 

## Type of Change

- [x] Feature
- [ ] Bug fix
- [ ] Documentation

## Contributors

- @Yanqiao4396 

## Reminder

 - All GitHub Actions should be in a passing state before any pull request is merged.
 - All PRs must be reviewed by at least one team member and one member of the Integration team!
 - Any issues this PR closes are tagged in the description!

### CODE REVIEW

**Comment by mariakimheinert** [comment]
Hi @Yanqiao4396, since this PR contains at least a small amount of source code, please request review from a member of the Integration team. Additionally, all PRs should be reviewed by another member of your own team, so you should also request review from a team member. You can see the [reviewer requirements on the course website](https://mariakimheinert.com/software-engineering/project/deliverables/individual-contributions/).

File: gatorgrade/generate/generate.py
Comment (laurennevill) [inline gatorgrade/generate/generate.py:4]: Hello @Yanqiao4396! These changes look good. The only thing I am noticing is your docstrings within your functions do not end with a `.`. The integration team plans to merge changes that include `pydocstyle` as a dependency which enforces these docstring standards. In order to ensure a passing build, you should add a period at the end of each of your docstrings.

**Review by laurennevill — CHANGES_REQUESTED** [review]

File: gatorgrade/generate/generate.py
Comment (laurennevill) [reply gatorgrade/generate/generate.py:4]: Additionally, you will need to have a docstring as the first line in the overall module for the same reasons as before.

Comment (Yanqiao4396) [reply gatorgrade/generate/generate.py]: Thanks for your review. I just updated and fixed the problems you talked about. Please check the new commit.

**Review by laurennevill — APPROVED** [review]

**Review by Nathandloria — APPROVED** [review]
This looks good! In the future, consider type hinting for the arguments and return values of the methods using Typer. This can occur later, I am just drawing your attention to it now.

File: gatorgrade/generate/generate.py
Comment (mariakimheinert) [inline gatorgrade/generate/generate.py:1]: Instead of "default messages and specific paths", could this docstring reflect the fact that this functionality will generate a GatorGrade configuration file with paths to whitelisted files and default GatorGrader checks?

File: gatorgrade/generate/generate.py
Comment (mariakimheinert) [inline gatorgrade/generate/generate.py:14]: For consistency, the "w" in "write" should be capitalized here.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Hi @Yanqiao4396, thanks for these important additions! I've left two points of feedback below. Please let me know if you have any questions.

File: gatorgrade/generate/generate.py
Comment (Michionlion) [inline gatorgrade/generate/generate.py:17]: In this stub, you may want to specify the parameters that this function will need -- it seems that at least a `path` parameter would be applicable.

In a PR such as this, it may be a good idea to stub out tests for these functions as well.

**Review by Michionlion — CHANGES_REQUESTED** [review]
This stub file looks good, but there are two possible issues: first, in the body of this PR you link to #46 as the "Linked Issue"; however, this number directs to this exact PR. The "Linked Issue" section is intended for issues that would be *closed* by this PR, meaning that the problem the issue describes is what the code in this PR would solve. So, you should change that link by either removing it, or linking to an issue that this PR resolves (maybe #44?).

Second, the `generate` folder needs an `__init__.py` file, since it should be a python package. This file does not need to contain any code, but does need to contain a docstring at the top of the file that documents the package as a whole. You can take a look at the `output` folder in [`main`](https://github.com/GatorEducator/gatorgrade/tree/main/gatorgrade/output) for a minimal example.

File: gatorgrade/generate/generate.py
Comment (Yanqiao4396) [reply gatorgrade/generate/generate.py]: Thanks for your time and proactive comments. I just did some improvements. Please re-review it.

**Review by laurennevill — APPROVED** [review]
Good work

**Review by Nathandloria — APPROVED** [review]

**Comment by mariakimheinert** [comment]
@Yanqiao4396, it seems that the only task remaining before this important PR can be merged is to have someone from your team approve it. @jackturner83 @anthonygc @Ochirsaikhan, could someone please review this PR and approve it if you believe these changes are ready to merge?

**Review by Ochirsaikhan — APPROVED** [review]

**Review by anthonygc — APPROVED** [review]
This looks pretty good! There does not seem to be any source code to begin with. However, I see that you have the basic outline of what you are going to do. It is very well documented and clean. Good job!

**Review by jackturner83 — APPROVED** [review]
Approve

**Pull request merged**
